c10/core/impl/DeviceGuardImplInterface.h

```diff
@@ -80,6 +81,45 @@ struct C10_API DeviceGuardImplInterface {
    */
   virtual Stream exchangeStream(Stream) const noexcept = 0;
 
+/**
+ * Destroys the given event.
+ */
+  virtual void destroyEvent (
+    void* event,
+    const DeviceIndex device_index) const noexcept = 0;
+
+/**
+ * Marks the event as not recorded and enqueues the event in the
+ * stream's work queue. When the stream processes the event either:
+ *  (1) the event is marked as recorded
+ *  (2) if the event was enqueued again, nothing happens
+ * Put another way, events reflect only the most recent call to record.
+ */
+  virtual void record(
+    void** event,
```

Comment:
Can this briefly explain why this is a `void**` rather than just a `void*`?

Reply:
record() may modify the void* event_ pointer since it might allocated a cuda or hip event. So calling it requires &event_ which is of type void**.